Assets/Scripts/RemoteInput.cs

```diff
@@ -117,12 +200,73 @@ public static void Reset()
             if (!m_isInitialized)
                 return;
 
+            foreach(int index in GamepadsMap.Values)
+            {
+                InputSystem.QueueStateEvent(Gamepad.all[index], new GamepadState());
+            }
             InputSystem.QueueStateEvent(RemoteMouse, new MouseState());
             InputSystem.QueueStateEvent(Keyboard, new KeyboardState());
             InputSystem.QueueStateEvent(RemoteTouch, new TouchState());
             InputSystem.Update();
         }
 
+#region Gamepads Events
+        static void ProcessGamepadButtonEvent(GamepadEventType state, GamepadKeyCode buttonIndex, double id, double value)
+        {
+            GamepadButton buttonToUpdate=GamepadButton.DpadUp;
+            GamepadState gamepadState = GamepadsStates[id];
+            switch(buttonIndex)
+            {
+                case GamepadKeyCode.dpadUp:
+                    buttonToUpdate = GamepadButton.DpadUp;
+                    break;
+                case GamepadKeyCode.dpadDown:
+                    buttonToUpdate = GamepadButton.DpadDown;
+                    break;
+                case GamepadKeyCode.dpadLeft:
+                    buttonToUpdate = GamepadButton.DpadLeft;
+                    break;
+                case GamepadKeyCode.dpadRight:
+                    buttonToUpdate = GamepadButton.DpadRight;
+                    break;
+                case GamepadKeyCode.Button0:
+                    buttonToUpdate = GamepadButton.B;
+                    break;
+                case GamepadKeyCode.Button1:
+                    buttonToUpdate = GamepadButton.A;
+                    break;
+                case GamepadKeyCode.Button2:
+                    buttonToUpdate = GamepadButton.Y;
+                    break;
+                case GamepadKeyCode.Button3:
+                    buttonToUpdate = GamepadButton.X;
+                    break;
+                case GamepadKeyCode.Button6:
+                    buttonToUpdate = GamepadButton.LeftTrigger;
+                    gamepadState.leftTrigger = (float)value;
+                    break;
+                case GamepadKeyCode.Button7:
+                    buttonToUpdate = GamepadButton.RightTrigger;
+                    gamepadState.rightTrigger = (float) value;
+                    break;    
+                default:
+                    UE.Debug.Log("Unmapped button code: " + buttonIndex);
+                    break;                   
+            }
+            GamepadsStates[id] = gamepadState.WithButton(buttonToUpdate, GamepadEventType.ButtonDown == state || GamepadEventType.ButtonPressed == state);
+        }
+
+        static void ProcessGamepadAxisEvent(double x, double y, GamepadKeyCode axisKeyCode, double id)
+        {
+            GamepadState gamepadState = GamepadsStates[id];
+            if(axisKeyCode == GamepadKeyCode.axis0)
+                gamepadState.leftStick = new UE.Vector2((float)x, (float)y);
```

Comment:
I guess it is better to use `float` type from [javascript side](https://github.com/Unity-Technologies/UnityRenderStreaming/pull/197/files#diff-aac2959976b324c8dc4fd1a33529706dR203-R209) because double parameters are cast here.

Reply:
Have you checked this? @gewentao

Reply:
I think all numbers in javascript are stored as 64 bits floating value so it's ok if we cast them to `float` in C# side.

Reply:
I think all numbers in javascript are stored as 64 bits floating value and the cast here is ok because float is enough to represent the axis value here.